### PR TITLE
feat: migrate chat system to PostgreSQL GraphQL schema (Chat Feature – Phase 1)

### LINKED ISSUE #2842: NoMongo: Migrate Chat System to New PostgreSQL-based GraphQL Schema ( Chat Feature – Phase 1 )
**Is your feature request related to a problem? Please describe.**
- The current chat system in Talawa mobile app is using the old MongoDB-based GraphQL schema which is incompatible with the new PostgreSQL-based backend. 

**Describe the solution you'd like**
- A comprehensive migration of the chat system to work seamlessly with the new PostgreSQL-based GraphQL schema while maintaining backward compatibility and improving user experience.

**Additional context**
- This issue is part of the Palisadoes internship program and covers the deliverables required for the first week, with additional bug fixes and improvements included:

- Week 1 Deliverables Completed:
✅ Chat System Migration: Successfully migrated from MongoDB to PostgreSQL schema
✅ Model Updates: Refactored all chat-related models for new data structure
✅ Query Migration: Updated GraphQL queries to work with new backend
✅ Test Updates: Ensured all tests pass with new implementation

- Bonus Improvements:
✅ App Tour Fixes: Resolved navigation conflicts and drawer closing issues
✅ UI Polish: Fixed FloatingActionButton overflow and improved chat UI
✅ Error Handling: Added comprehensive error handling throughout the system
✅ Type Safety: Improved code quality with better type annotations
✅ Test Coverage: Achieved 100% line coverage for chat components

### PR DESCRIPTION
**Pull request by may-tas**
### What kind of change does this PR introduce?

- Migration of some of the chat features to work with the new PostgreSQL-based GraphQL schema.

### Issue Number:

- Fixes #2842 

### Did you add tests for your changes?

- [x] Tests are written for all changes made in this PR.
- [x] Test coverage meets or exceeds the current coverage (~90/95%).

## Core Changes
- **GraphQL Schema Migration**: Updated queries to use new `chat` and `usersByOrganizationId` endpoints
- **Model Refactoring**: Removed JsonSerializable, implemented custom fromJson/toJson methods
- **Field Mapping**: Updated field names (body→messageContent, creator→sender, name→firstName)
- **Backward Compatibility**: Support both old and new field formats during transition

## Enhanced Features
- **ChatMessage Model**: Dynamic sender/receiver logic based on current user context
- **ChatUser Model**: Smart name splitting and avatarURL handling
- **User Model**: Support for both nested and direct data structures
- **Service Layer**: Comprehensive error handling and type safety improvements

## UI/UX Improvements
- **App Tour**: Fixed navigation flow with proper drawer closing sequence
- **Select Contact**: Enhanced with loading states, error handling, and avatar fallbacks
- **Chat List**: Resolved FloatingActionButton overflow issues
- **Loading States**: Added proper busy states and progress indicators

### Does this PR introduce a breaking change?

- No

### Checklist for Repository Standards
- [x] Have you reviewed and implemented all applicable `coderaabbitai` review suggestions?
- [x] Have you ensured that the PR aligns with the repository’s contribution guidelines?

### Other information

- Part of: Palisadoes internship Week 1 deliverables + bonus improvements

### Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?

- Yes

## Summary by CodeRabbit

* **New Features**
  * Added loading and empty state handling to the contact selection screen.
  * Re-enabled chat functionality in the main navigation and updated tutorial flow to include chat earlier.
  * Added a new GraphQL query for fetching users by organization.

* **Improvements**
  * Enhanced error handling and robustness in organization member and chat message fetching.
  * Improved avatar and name display logic for chat contacts.
  * Refactored chat and user models for manual JSON serialization, supporting multiple data formats.
  * Expanded and clarified documentation across models and query utilities.

* **Bug Fixes**
  * Fixed issues with avatar fallback logic and user name display in chat contact lists.
  * Improved error handling and state management in contact selection and chat-related view models.

* **Tests**
  * Added comprehensive widget and unit tests for chat screens, contact selection, and chat/user models.
  * Updated and expanded tests for organization services, event models, and main screen navigation.
  * Improved test coverage for error scenarios and edge cases across multiple modules.
  * Reformatted and clarified test code for better readability and maintainability.

* **Refactor**
  * Removed reliance on code generation for chat and user models, switching to manual serialization.
  * Updated method signatures and internal logic for clarity and maintainability.

* **Style**
  * Improved code formatting and added detailed documentation comments throughout the codebase.

* **Chores**
  * Updated service locator and mock setups to support new and refactored features.
  * Removed obsolete generated files and outdated tests.

### CODE REVIEW

File: lib/views/after_auth_screens/chat/select_contact.dart [hunk 1]
```diff
@@ -40,23 +41,55 @@ class _SelectContactState extends State<SelectContact> {
...
           return ListView.builder(
             itemCount: model.orgMembersList.length,
             itemBuilder: (context, index) {
+              final user = model.orgMembersList[index];
               return GestureDetector(
                 key: Key('select_contact_gesture_$index'),
                 onTap: () {
-                  // Navigator.push(context, MaterialPageRoute(builder: (context)=>ChatMessageScreen(chat: ChatListTileDataModel(ChatUser(model.orgMembersList[index].firstName,model.orgMembersList[index].id,model.orgMembersList[index].image),null,0))));
+                  // TODO: Navigate to chat screen with selected user
+                  debugPrint('Selected user: ${user.firstName} (${user.id})');
```
[the remark this answers is not in the document]
Comment (may-tas) [reply lib/views/after_auth_screens/chat/select_contact.dart:66]: This TODO is intended and will be implemented in the next PR.

File: lib/utils/chat_queries.dart [hunk 2]
```diff
@@ -20,29 +22,55 @@ class ChatQueries {
...
+                endCursor
+                hasNextPage
+                hasPreviousPage
+                startCursor
+              }
             }
           }
         }
     ''';
   }
```
[the remark this answers is not in the document]
Comment (may-tas) [reply lib/utils/chat_queries.dart:65]: This limit is intended, as the message screen is not yet implemented and tested, this change will be included in the next PR.

File: test/views/after_auth_screens/chat/chat_list_screen_test.dart [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+      final iconFinder = find.descendant(
+        of: fabFinder,
+        matching: find.byType(Icon),
+      );
+      expect(iconFinder, findsOneWidget);
+
+      final Icon icon = tester.widget<Icon>(iconFinder);
+      expect(icon.icon, Icons.contacts);
+      expect(icon.color, Colors.white);
+    });
```
[the remark this answers is not in the document]
Comment (may-tas) [reply test/views/after_auth_screens/chat/chat_list_screen_test.dart:172]: I think that is excessive and not much required.

File: test/widget_tests/pre_auth_screens/select_language_page_test.dart [hunk 4]
```diff
@@ -281,10 +285,61 @@ Future<void> main() async {
...
+
+      // Verify navigation to app settings page
+      verify(
+        mockNavigationService.popAndPushScreen(
+          '/appSettingsPage',
+          arguments: '',
+        ),
+      ).called(1);
+    });
+  });
```
[the remark this answers is not in the document]
Comment (may-tas) [reply test/widget_tests/pre_auth_screens/select_language_page_test.dart:344]: That is enough for now, as its an additional test case to complete the code coverage.

**Comment by may-tas** [comment]
@coderabbitai review

**Comment by may-tas** [comment]
@coderabbitai Re-request review

**Comment by may-tas** [comment]
@noman2002 Kindly review it, I have made all the changes requested by **coderabbitai** and all checks are passing.

**Comment by noman2002** [comment]
LGTM 
@Dante291  please a look.

**Review by noman2002 — APPROVED** [review]

**Comment by Dante291** [comment]
@may-tas can you please add a screen recording of chat working fine now in app.

**Comment by may-tas** [comment]
> @may-tas can you please add a screen recording of chat working fine now in app.

@Dante291 

This is still a work in progress. I've restored some UI features and updated several queries, mutations, and models. Most of the changes in this PR are related to backend updates. In the next PR, I'll focus on connecting all the queries and mutations with the UI and continue completing the remaining parts step by step.

https://github.com/user-attachments/assets/d6edf50b-b15b-4900-b1dd-67f1055d6b65
